lib/plugins/aws/invokeLocal/index.js

```diff
@@ -443,113 +417,111 @@ class AwsInvokeLocal {
       'invokeLocal',
       'artifact'
     );
-    return decompress(artifact, destination).then(() => destination);
+    await decompress(artifact, destination);
+    return destination;
   }
 
   getEnvVarsFromOptions() {
-    const envVarsFromOptions = {};
     // Get the env vars from command line options in the form of --env KEY=value
-    [].concat(this.options.env || []).forEach(itm => {
-      const splitItm = itm.split('=');
-      envVarsFromOptions[splitItm[0]] = splitItm.slice(1, splitItm.length).join('=') || '';
-    });
-    return envVarsFromOptions;
+    return []
+      .concat(this.options.env || [])
+      .map(itm => itm.split('='))
+      .map(([varName, ...varValue]) => [varName, varValue.join('=')])
+      .reduce(
+        (envVarsFromOptions, [varName, varValue]) => ({
+          ...envVarsFromOptions,
+          [varName]: varValue || '',
+        }),
+        {}
+      );
   }
 
-  ensurePackage() {
-    if (this.options['skip-package']) {
-      return fse
-        .access(
-          path.join(this.serverless.config.servicePath, '.serverless', 'serverless-state.json')
-        )
-        .catch(() => this.serverless.pluginManager.spawn('package'));
+  async ensurePackage() {
+    if (
+      this.options['skip-package'] &&
+      (await fileExists(
```

Comment:
`fileExist` can not crash because its implementation is:

```js
  return fse
    .lstat(filePath)
    .then(stats => stats.isFile())
    .catch(() => false);
```

It is one of those extension methods that you have especially for this case.

Reply:
That's true, still my point was that we should maintain original logic (I overlooked you replaced `fs.access` with `fileExists`).

We should keep`fse.access` (it does a bit more that `fse.exists`).

Ideally this PR should be just about async/await refactor, nothing more (let's keep different concerns separated)

Reply:
Reverted to `fse.access` and previous logic.